### PR TITLE
Feature/sliding pane

### PR DESCRIPTION
**Pull request by abika**
As promised in #371 here is a new layout for the main view: The contact list is now a sliding side view for the conversation list.

Details:
* most code in the contacts activity was moved to either the conversation (list) activity or to the fragment
* the conversation activity contains now both fragments, for conversation and contact list
* the contacts list activity is still there but only for picking a contact (invitation is not possible here anymore, but whatever)

Tested with normal and dual-pane view + orientation change in every possible situation.
[Sometimes the syncing process takes very long (60sec). Annoying in the view, but that's another problem]

Maybe a nice TODO: show the slider from the right side. But I couldn't found an easy solution for that.

### CODE REVIEW

**Comment by h-2** [comment]
Isnt there a redesign with #412 anyway? [but sounds nice]

**Comment by TheLastProject** [comment]
Any chance for a screenshot so non-developers could comment? If you are okay with that, of course.

**Comment by abika** [comment]
@h-2 true, but I didn't want to merge with another feature branch

@TheLastProject nothing much to see. In the conversation list view you click the "new conversation" icon, the contact list slides in from the left side. Choose a contact, swipe back or click the "back" button to go back.
device-2015-08-31-205819

**Comment by daniele-athome** [comment]
Thanks @abika that's great work! I'll begin reviewing it and merging it to master ASAP.
By the way do you have any proposal for merging this to the material branch? I mean do you think still using a floating action button would be ideal in this situation? Any other UI insight might be useful.

**Comment by abika** [comment]
I created another branch merging the sliding pane with material design (3171992f0c2d2a48137f785f0f389a5aa52d60d6). [The layout gets a bit complicated with the new toolbar]

> I mean do you think still using a floating action button would be ideal in this situation? Any other UI insight might be useful.

A button is needed here, we can even disable swiping in the conversation view to force using the button.

But honestly, I thought the old button in the toolbar was fine. It was more unobtrusive, which is good as creating new conversations will happen seldom if the app is used for a while (the new icon is better though).

Besides that the new design looks really great btw.! Can't thing of any improvements

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -101,9 +101,10 @@
             android:value=".ui.SearchActivity" />
 
         <!-- Activities -->
-        <activity android:name=".ui.ConversationList"
+        <activity android:name=".ui.ConversationActivity"
             android:launchMode="singleTop"
-            android:theme="@style/Theme.Light">
+            android:theme="@style/Theme.Light"
+            android:configChanges="orientation|screenSize">
```
Comment (daniele-athome) [inline app/src/main/AndroidManifest.xml]: Any particular reason for adding this configChanges?

Comment (abika) [inline app/src/main/AndroidManifest.xml]: yes: if the contact list is open and the screen rotates, the action bar will be reset to the settings for the conversion list. These configChanges  prevent this by simply ignoring the screen changes.

Comment (daniele-athome) [inline app/src/main/AndroidManifest.xml]: Configuration changes can lead to some unexpected results when are not handled by the system (e.g. resources are not reloaded). I'll adjust the code to make it happen automatically according to the activity lifecycle.

Comment (abika) [inline app/src/main/AndroidManifest.xml]: ok thx, I didn't know how to do that. In the onCreate() call caused by the screen rotation the sliding pane was set to be closed even though it was visibly open.

**Pull request merged**